src/java.desktop/share/classes/java/awt/TextComponent.java

```diff
@@ -359,6 +359,11 @@ public void setBackground(Color c) {
         super.setBackground(c);
     }
 
+    void setBackground(Color c, boolean setByClient) {
+        backgroundSetByClientCode = setByClient;
+        super.setBackground(c);
+    }
+
```

Comment:
Does this need a method need a javadoc comment block like the `setBackground` method before it or like any of the methods in the class?

Reply:
since this method isn't a public or protected method (it's package-private, so it can't be accessed by users) there's no need for a javadoc comment

Reply:
Right. OK I'll wait for you to move the test to the AWT directory then because I do agree, the components involved seem to be AWT rather than Swing ones.